docker-compose.yml

```diff
@@ -1,69 +1,128 @@
-# NOTE: This docker-compose.yml is meant to be just an example of how
-# you could accomplish this on your own. It is not intended to work in
-# all use-cases and must be adapted to fit your needs. This is merely
-# a guideline.
-
-# See docs.getsentry.com/on-premise/server/ for full
-# instructions
-
 version: '3.4'
-
-x-defaults: &defaults
+x-defaults:
   restart: unless-stopped
-  build:
+  build: &ref_0
     context: .
     args:
       SENTRY_IMAGE: ${SENTRY_IMAGE}
-  depends_on:
+  depends_on: &ref_1
     - redis
     - postgres
     - memcached
     - smtp
+    - snuba
+    - symbolicator
+    - kafka
   env_file: .env
-  environment:
+  environment: &ref_2
     SENTRY_MEMCACHED_HOST: memcached
     SENTRY_REDIS_HOST: redis
     SENTRY_POSTGRES_HOST: postgres
     SENTRY_EMAIL_HOST: smtp
-  volumes:
-    - sentry-data:/var/lib/sentry/files
-
-
+    SNUBA: 'http://snuba:1218'
+    SENTRY_KAFKA_HOST: 'kafka:9092'
+  volumes: &ref_3
+    - 'sentry-data:/var/lib/sentry/files'
 services:
   smtp:
     restart: unless-stopped
     image: tianon/exim4
-
   memcached:
     restart: unless-stopped
-    image: memcached:1.5-alpine
-
+    image: 'memcached:1.5-alpine'
   redis:
     restart: unless-stopped
-    image: redis:3.2-alpine
-
+    image: 'redis:5.0-alpine'
+    command: redis-server --appendonly yes
+    volumes:
+      - 'sentry-redis:/data'
   postgres:
     restart: unless-stopped
-    image: postgres:9.5
+    image: 'postgres:9.6-alpine'
```

Comment:
I'd note, this change is a backwards incompatible change. Anyone that was running Sentry is going to have a bad time.

Reply:
I need to dig into the upgrade process then, good point. 9.6-alpine is the only one free of vulnerabilities btw, according to this page: https://hub.docker.com/_/postgres?tab=tags (also 15MB vs 83MB, 💥 )

Reply:
Oh, I also missed the change to alpine. alpine is not great for Postgres, so let's not do that.

My main concern is from 9.5 -> 9.6 since it requires pg_upgrade to be run to work.

Reply:
Well then, I'm happy to keep the old one as the main reason was for alpine and smaller footprint.

Reply:
I'd prefer if we get to 9.6 since that's what we runin productino ourselves, I was just noting that the change from 9.5 -> 9.6 requires an explicitly pg_upgrade run.

The change to alpine is another can of worms I'm not opening.